File: SlidingWindow/Subarraysum.java
public class Subarraysum {

    // Function to find a subarray with a given sum
    public static int[] findSubarraySum(int[] arr, int targetSum) {
        int left = 0;
        int right = 0;
        int currentSum = 0;

        while (right < arr.length) {
            currentSum += arr[right];

            while (currentSum > targetSum) {
                currentSum -= arr[left];
                left++;
            }

            if (currentSum == targetSum) {
                // Subarray found, return it
                int[] subarray = new int[right - left + 1];
                System.arraycopy(arr, left, subarray, 0, right - left + 1);
                return subarray;
            }

            right++;
        }

        // If no subarray is found with the targetSum
        return new int[]{};
    }

    public static void main(String[] args) {
        int[] arr = {1, 2, 3, 4, 5};
        int targetSum = 9;

        int[] result = findSubarraySum(arr, targetSum);

        if (result.length > 0) {
            System.out.print("Subarray with sum " + targetSum + ": ");
            for (int num : result) {
                System.out.print(num + " ");
            }
        } else {
            System.out.println("No subarray found with the target sum.");
        }
    }
}

File: SlidingWindow/SubarraySum.py
def subarray_sum(arr, target_sum):
    window_sum = 0
    left, right = 0, 0

    while right < len(arr):
        window_sum += arr[right]

        while window_sum > target_sum:
            window_sum -= arr[left]
            left += 1

        if window_sum == target_sum:
            return arr[left:right + 1]

        right += 1

    return None  # If no subarray is found with the target_sum

arr = [1, 2, 3, 4, 5]
target_sum = 9
result = subarray_sum(arr, target_sum)
print(result)  # Output: [2, 3, 4]
